deploy_preservica_config.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  preservica_config_bucket_name = "${local.environment}-dr2-preservica-config"
+  preservica_config_lambda_name = "${local.environment}-preservica-config"
+}
+module "preservica_config_bucket" {
+  source      = "git::https://github.com/nationalarchives/da-terraform-modules//s3"
+  bucket_name = local.preservica_config_bucket_name
+  sns_topic_config = {
+    "s3:ObjectCreated:*" = module.preservica_config_sns.sns_arn
+  }
+  kms_key_arn = module.dr2_developer_key.kms_key_arn
+  bucket_policy = templatefile("${path.module}/templates/s3/preservica_config_bucket_policy.json.tpl", {
+    preservica_config_lambda_role_arn = module.preservica_config_lambda.lambda_role_arn
+    bucket_name                       = local.preservica_config_bucket_name
+  })
+  logging_bucket_policy = templatefile("${path.module}/templates/s3/log_bucket_policy.json.tpl", {
+    bucket_name = "${local.preservica_config_bucket_name}-logs", account_id = var.dp_account_number
+  })
+}
+
+module "preservica_config_sns" {
+  source = "git::https://github.com/nationalarchives/da-terraform-modules//sns"
+  sns_policy = templatefile("${path.module}/templates/sns/s3_notifications_policy.json.tpl", {
+    topic_name  = "${local.environment}-preservica-config"
+    bucket_name = "${local.environment}-dr2-preservica-config"
+    account_id  = data.aws_caller_identity.current.account_id
+  })
+  tags = {
+    Name = "Preservica Config SNS"
+  }
+  topic_name  = "${local.environment}-preservica-config"
+  kms_key_arn = module.dr2_developer_key.kms_key_arn
+  sqs_subscriptions = {
+    preservica_config_queue = module.preservica_config_queue.sqs_arn
+  }
+}
+
+module "preservica_config_queue" {
+  source     = "git::https://github.com/nationalarchives/da-terraform-modules//sqs"
+  queue_name = "${local.environment}-preservica-config"
+  sqs_policy = templatefile("${path.module}/templates/sqs/sns_send_message_policy.json.tpl", {
+    account_id = data.aws_caller_identity.current.account_id, queue_name = "${local.environment}-preservica-config"
+    queue_name = "${local.environment}-preservica-config"
+    topic_arn  = module.preservica_config_sns.sns_arn
+  })
+  kms_key_id         = module.dr2_developer_key.kms_key_arn
+  visibility_timeout = 60
```

Comment:
> To allow your function time to process each batch of records, set the source queue's [visibility timeout](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html) to at least six times the [timeout that you configure](https://docs.aws.amazon.com/lambda/latest/dg/configuration-function-common.html#configuration-common-summary) on your function.

https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-queueconfig:~:text=To%20allow%20your%20function%20time%20to%20process%20each%20batch%20of%20records%2C%20set%20the%20source%20queue%27s%20visibility%20timeout%20to%20at%20least%20six%20times%20the%20timeout%20that%20you%20configure%20on%20your%20function.

Reply:
Fair enough, I've set it to 360.